lib/airbyte/config_api/v1/source_definition.rb

```diff
@@ -0,0 +1,50 @@
+
+module Airbyte
+  def self.source_definition; SourceDefinition.new; end
+  class SourceDefinition < ConfigAPIClient
+    def list()
+      handle_request("#{PATH_PREFIX_SOURCE_DEFINITIONS}/list")
+    end
+
+    def get_id(source_name)
+      find_in_definitions(source_name, list())
+    end
+
+    def list_for_workspace(workspace_id)
+      params = {
+        workspaceId: workspace_id
+      }
+      handle_request("#{PATH_PREFIX_SOURCE_DEFINITIONS}/list_for_workspace", body: params)
+    end
+
+    def get_id_for_workspace(source_name, workspace_id)
+      find_in_definitions(source_name, list_for_workspace(workspace_id))
+    end
+
+    def add_custom_definition_for_workspace(workspace_id, definition_info)
```

Comment:
where this endpoint is going to be used ? Is it for databricks and other custom source ?

Reply:
It is only for custom sources, it will used once per workspace ie: if we keep one workspace per site then it will be called during first adaptor creation.